dcc-download-server/src/main/java/org/icgc/dcc/download/server/endpoint/DownloadController.java

```diff
@@ -105,24 +97,41 @@ public void downloadArchive(
     streamArchive(Optional.of(jobId), streamerOpt, response);
   }
 
+  public void downloadArchiveInfo(@RequestParam("token") String token,
+      @RequestParam(value = "type", required = false) String dataType,
+      @NonNull HttpServletResponse response) throws IOException {
+    val jobId = processDynamicDownloadToken(token);
+
+    val downloadDataType = getDownloadDataType(dataType);
+    val fileStreamer = downloadDataType.isPresent() ?
+        downloadService.getNoDataDynamicFileStreamer(jobId, downloadDataType.get()) :
+        downloadService.getNoDataDynamicFileStreamer(jobId);
+
+    // Will not stream anything as the fileStreamer is a non-streaming one.
+    streamArchive(Optional.of(jobId), fileStreamer, response);
+  }
+
   @CrossOrigin(origins = "*")
   @RequestMapping(value = "/static", method = GET)
   public void staticDownload(@RequestParam("token") String token, HttpServletResponse response) throws IOException {
     log.debug("Received download request. Token: '{}'", token);
-    val tokenPayload = getTokenPayload(token);
-
-    val requestPath = tokenPayload.getPath();
-    checkArgument("path", requestPath);
-    log.debug("Static path: '{}'", requestPath);
-
-    val filePath = getFsPath(requestPath);
+    val filePath = getStaticDownloadFilePath(token);
     log.info("Getting download archive for path '{}'", filePath);
 
     val output = response.getOutputStream();
     val streamerOpt = downloadService.getStaticArchiveStreamer(filePath, output);
     streamArchive(Optional.empty(), streamerOpt, response);
   }
 
+  @CrossOrigin(origins = "*")
```

Comment:
Can put this on top of class:

https://github.com/icgc-dcc/dcc-dev/blob/develop/dcc-dev-server/src/main/java/org/icgc/dcc/dev/server/portal/PortalController.java#L53